measureme/src/profiler.rs

```diff
@@ -122,6 +122,41 @@ impl Profiler {
         }
     }
 
+    /// Creates a "start" event and returns a `DetachedTiming`.
+    /// To create the corresponding "event" event, yuu must call
+    /// `finish_recording_internal_event` with the returned
+    /// `DetachedTiming`
+    #[inline]
+    pub fn start_recording_interval_event_detached(
+        &self,
+        event_kind: StringId,
+        event_id: EventId,
+        thread_id: u32
+    ) -> DetachedTiming {
+        DetachedTiming {
+            event_id,
+            event_kind,
+            thread_id,
+            start_count: self.counter.since_start(),
+        }
+    }
+
+    /// Creates the corresponding "end" event for
+    /// the "start" event represented by `timing`. You
+    /// must have obtained `timing` from the same `Profiler`
+    pub fn finish_recording_interval_event(
+        &self,
+        timing: DetachedTiming
+    ) {
+        drop(TimingGuard {
+            profiler: self,
+            event_id: timing.event_id,
+            event_kind: timing.event_kind,
+            thread_id: timing.thread_id,
+            start_count: timing.start_count,
+        });
+    }
```

Comment:
Would

```rust
let start_count = profiler.counter.since_start();
// some work
let raw_event = RawEvent::new_interval(
    event_kind,
    event_id,
    thread_id,
    start_count,
    profiler.counter.since_start(),
);
profiler.record_raw_event(&raw_event);
```

work if you make `profiler.counter.since_start()` public? That is what the drop impl of `TimingGuard` does.

Reply:
It would, but I think this API is more consistent with the existing one (hiding the underlying counter). I don't have a strong opinion, though.